src/core/math.d

```diff
@@ -40,7 +49,9 @@ public:
  *      Results are undefined if |x| >= $(POWER 2,64).
  */
 
-real cos(real x) @safe pure nothrow;       /* intrinsic */
+float cos(float x);     /* intrinsic */
```

Comment:
Shouldn't this be `cosf` and the `real` be `cosl`? Same throughout.

Reply:
intrinsics are detected based on the mangled name https://github.com/dlang/dmd/pull/7995/files#diff-34b7829e9285bc4fe4e718f0653a6115R421
so no.